### PR TITLE
Feat/dl input

### PR DESCRIPTION
**Pull request by prodan-eric**
New input component with suggestions, images, files that can be expandable using a prop

### CODE REVIEW

**Comment by fadiDL** [comment]
@SorinOpt @prodan-eric this is missing the text-area feature where if you input alot of text that is longer than the width and has the prop expandable then  it grows in height as you write more text. just like in slack.

**Comment by prodan-eric** [comment]
> @SorinOpt @prodan-eric this is missing the text-area feature where if you input alot of text that is longer than the width and has the prop expandable then it grows in height as you write more text. just like in slack.

The prop is there, it's called "expandable". The second example in the demo showcases it 
image

**Comment by fadiDL** [comment]
> > @SorinOpt @prodan-eric this is missing the text-area feature where if you input alot of text that is longer than the width and has the prop expandable then it grows in height as you write more text. just like in slack.
> 
> The prop is there, it's called "expandable". The second example in the demo showcases it image

re testing now 👍🏼

**Comment by fadiDL** [comment]
Looking great @prodan-eric can you add one missing option that the suggestions have an extra property on the object passed called click
if this property is preset then the suggestion can be clicked and it will trigger the function. (on hover will show underline like a link and mouse changes to pointer)

**Comment by fadiDL** [comment]
@prodan-eric you misunderstood me with the clickable property, its my bad.

We need to be able to click on the suggestion after its selected so on
image

the blue text should be clickable and emit the event not the suggestion list

Furthermore, i see that you started attempting to implement ellipsis on the input content. 
Please notify me to merge this PR when all is ready

**Comment by fadiDL** [comment]
@prodan-eric ive went ahead and changed min-width prop to width even tho in the css its still min width to not break behavior 

the reason for this is the input should flow into all the container by default unless given specific width / height

The following is a styling issue encountered after its because of the way the new input is built. can you please take a look ? 

image

**Comment by fadiDL** [comment]
@prodan-eric 
image

the icons are not centered

File: src/components/compound/DlSearches/DlSmartSearch/components/DlSmartSearchInput.vue [hunk 1]
```diff
@@ -349,7 +349,7 @@ export default defineComponent({
             setInputValue(
                 clearPartlyTypedSuggestion(input.value.innerText, stringValue)
             )
-            setCaret(input.value)
+            setCaretAtTheEnd(input.value)
```
Comment (fadiDL) [inline src/components/compound/DlSearches/DlSmartSearch/components/DlSmartSearchInput.vue:352]: @prodan-eric why set at end ? 
we need to also modify the code in the middle. would this throw the cursor to the end aswell ?

File: src/components/compound/DlSearches/DlSmartSearch/utils/highlightSyntax.ts [hunk 2]
```diff
@@ -110,7 +110,7 @@ function renderText(text: string, colorSchema: SyntaxColorSchema) {
     return output?.join('')
 }
 
-export function setCaret(target: HTMLElement) {
+export function setCaretAtTheEnd(target: HTMLElement) {
```
Comment (fadiDL) [inline src/components/compound/DlSearches/DlSmartSearch/utils/highlightSyntax.ts:113]: i dont believe this is correct naming as descrbied in comment above

File: src/stories/DlInput.stories.js
Comment (fadiDL) [inline src/stories/DlInput.stories.js:1]: rewrite this to ts please. there is no need to keep old stories formats. we are slowly rewriting them to the new storybook format

**Review by fadiDL — CHANGES_REQUESTED** [review]

File: src/components/compound/DlSearches/DlSmartSearch/components/DlSmartSearchInput.vue (hunk 1 above)
Comment (fadiDL) [reply src/components/compound/DlSearches/DlSmartSearch/components/DlSmartSearchInput.vue:352]: @prodan-eric are we using the new selection util also in the smart search ?

Comment (prodan-eric) [reply src/components/compound/DlSearches/DlSmartSearch/components/DlSmartSearchInput.vue:352]: No, but I think it would be usable there

Comment (fadiDL) [reply src/components/compound/DlSearches/DlSmartSearch/components/DlSmartSearchInput.vue:352]: @prodan-eric its ok, a task for another day i believe

**Comment by fadiDL** [comment]
@prodan-eric it broke again .. somehow :( 
image

**Review by fadiDL — APPROVED** [review]

**Pull request merged**